src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1270,7 +1286,46 @@ spanByRolling
     -> Fold m a b
     -> Fold m a c
     -> Fold m a (b, c)
-spanByRolling cmp fld1 fld2 = splitWith (,) (groupByRolling cmp fld1) fld2
+spanByRolling cmp (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
```

Comment:
Make this a parser operation.

Reply:
Yes, you left this comment earlier (in the closed PR). I've replied by saying: This is a correct implementation and behavior so we might as well keep it.

Reply:
We don't want to maintain multiple implementations unless it gives us perf advantage over the parser version.

Reply:
The advantage is, if the use case can be solved using folds there is no need for the users to look into parsers.